src/AutoRest.CSharp/Common/Generation/Writers/ModelWriter.cs

```diff
@@ -231,40 +234,77 @@ private static void WriteGetWithEscape(CodeWriter writer, FlattenedObjectTypePro
 
         private FormattableString CreatePropertyDescription(ObjectTypeProperty property, string? overrideName = null)
         {
-            FormattableString binaryDataExtraDescription = CreateBinaryDataExtraDescription(property.Declaration.Type, property.SerializationFormat);
+            FormattableString binaryDataExtraDescription = CreateBinaryDataExtraDescription(property.InputModelProperty, property.Declaration.Type, property.SerializationFormat);
             if (!string.IsNullOrWhiteSpace(property.PropertyDescription.ToString()))
             {
                 return $"{property.PropertyDescription}{binaryDataExtraDescription}";
             }
             return $"{ObjectTypeProperty.CreateDefaultPropertyDescription(overrideName ?? property.Declaration.Name, property.IsReadOnly)}{binaryDataExtraDescription}";
         }
 
-        private FormattableString CreateBinaryDataExtraDescription(CSharpType type, SerializationFormat serializationFormat)
+        /// <summary>
+        /// This method creates the extra description for binary data properties. If the property represents a set of union types, it will attempt to
```

Comment:
Typically we want the Writers to be very dumb and the TypeProvider should present exactly what should be written.  We should move any summary logic into the class responsible for calculating the summary so that we can simply read it from the provider.

Also it is an anti-pattern for the writters to consume the input types, they should get all of their information from the output types.  I know this does accidentally exist elsewhere but we should not add more of them if we can help it.

Reply:
I moved this logic outside of the writers. Let me know if it's still not in the correct place.